src/realm/set.hpp

```diff
@@ -843,46 +795,59 @@ void Set<T>::do_erase(size_t ndx)
     m_tree->erase(ndx);
 }
 
+namespace {
 template <class T>
-template <class Rhs>
-bool Set<T>::is_subset_of(const Rhs& rhs) const
+auto convert_to_set(const CollectionBase& rhs, bool nullable)
 {
-    return is_subset_of(std::begin(rhs), std::end(rhs));
+    std::set<T, SetElementLessThan<T>> ret;
+    for (size_t i = 0; i < rhs.size(); i++) {
+        auto val = rhs.get_any(i);
+        if (val.is_type(ColumnTypeTraits<T>::id)) {
```

Comment:
I don't think this conversion function works when T=Mixed?

Reply:
Well, I should have known by now - shouldn't I?